src/main/java/sirius/biz/util/ArchiveHelper.java

```diff
@@ -103,4 +106,64 @@ private static void initSevenZipLib() throws SevenZipNativeInitializationExcepti
         return indices;
     }
 
+    /**
+     * Provides a progress of a extraction process.
+     */
+    public static class ExtractionProgress {
+
+        private final ExtractOperationResult extractOperationResult;
+        private final ByteSource data;
+        private final String filePath;
+        private final long filesProcessedSoFar;
+        private final long bytesProcessedSoFar;
+        private final long totalBytes;
+
+        /**
+         * Creates a new instance for progress status updates.
+         *
+         * @param extractOperationResult the extraction result of the current file
+         * @param data                   the data of the extracted file
+         * @param filePath               the temp file path to the extracted file on disk
+         * @param filesProcessedSoFar    the number of files processed so far
+         * @param bytesProcessedSoFar    the number of bytes processed so far
+         * @param totalBytes             the number of total bytes the archive has
+         */
+        public ExtractionProgress(ExtractOperationResult extractOperationResult,
+                                  ByteSource data,
+                                  String filePath,
+                                  long filesProcessedSoFar,
+                                  long bytesProcessedSoFar,
+                                  long totalBytes) {
+            this.extractOperationResult = extractOperationResult;
+            this.data = data;
+            this.filePath = filePath;
+            this.filesProcessedSoFar = filesProcessedSoFar;
+            this.bytesProcessedSoFar = bytesProcessedSoFar;
+            this.totalBytes = totalBytes;
+        }
+
+        public ExtractOperationResult getExtractOperationResult() {
+            return extractOperationResult;
+        }
+
+        public ByteSource getData() {
+            return data;
+        }
+
+        public String getFilePath() {
+            return filePath;
+        }
+
+        public long getFilesProcessedSoFar() {
+            return filesProcessedSoFar;
+        }
+
+        public long getBytesProcessedSoFar() {
```

Comment:
Rename to just getFilesProcessed?

is a progress, I guess "soFar" is somewhat redundant...